src/transformers/trainer.py

```diff
@@ -1867,6 +1868,7 @@ def _inner_training_loop(
 
             step = -1
             for step, inputs in enumerate(epoch_iterator):
+                total_steps += 1
```

Comment:
hmm, but it's not `total_samples` as each pull from a DL can be more than one sample if bs>1.

Reply:
also I think this bug will still be there for resumes unless the total counter is set to a saved at the end of the previous session value.

It's insignificant in the large scale, but it's likely to lead to a spike in the loss and other values on resume since the calculation of the averaged values will be wrong when divided by GAS.

Reply:
Or `total_batches` if you prefer.

Reply:
As for resuming, the resuming necessarily happens at a training step, so a round multiple of number of gradient accumulation batches. Therefore there should be no problem.

Reply:
> Or `total_batches` if you prefer.

I thought of this one, but I think it too is ambiguous due to GAS. 

May be `total_batched_samples`? That would be the closest I think if we want to speak DL and not steps.

a. `total_across_epochs_steps`
b. `total_batched_samples`
c. ???

Reply:
`total_batched_samples` works for me too. I just want to avoid steps as I know many people assume it's the number of `optimizer.step()` done.

Reply:
I hear you - totally, let's go with `total_batched_samples` then.